addons/edi_stock/models/edi_pick_report_document.py

```diff
@@ -153,6 +153,7 @@ def prepare(self, doc):
         Move = self.env['stock.move']
         # Lock pickings to prevent concurrent report generation attempts
         picks = Picking.search(self.pick_report_domain(doc), order='id')
+        picks = self.filter_picks_with_issues(picks, doc)
```

Comment:
nit: We may as well just do filter_picks then we can use it for other things in the future uf needed

Reply:
Agree